### ISSUE TITLE
Zoom Enabled Does Not Update With State

### ISSUE THREAD

**Issue by wakaztahir** [Tlaster/Zoomable#10]
As you can see I have a Box with Surface and a Canvas inside it and panning is not working , zooming is working and my drawing is displaying , I had a previous issue of zoom expanding the surface of canvas over other views which I fixed by clip to bounds modifier you can see on Zoomable

Getting rid of that modifier does not cause panning to work but instead that issue returns , I am not sure if this is an issue with your library or Compose works like this

**Comment by wakaztahir** [comment]
I also removed Box and Surface , But panning still didn't work on canvas !

**Comment by Tlaster** [comment]
It might because that Canvas consumes the position changes and Zoomable can not receive these changes, can you share more complete code of it?

**Comment by wakaztahir** [comment]
![Capture](https://user-images.githubusercontent.com/42442700/123056138-09990f80-d420-11eb-85ff-f75e29068c2f.PNG)
![Capture2](https://user-images.githubusercontent.com/42442700/123056142-0aca3c80-d420-11eb-9eca-af4d806baae5.PNG)

**Comment by wakaztahir** [comment]
returning true to pointerInteropFilter , returning false in else condition , tell me if I need to change these !

**Comment by wakaztahir** [comment]
if I return false in every all actions , it does not consume event and does not draw , if I return true in else condition it does not make a difference

Zoomable is parent of Sketcher Composable , should not parent consume pointer events first and then pass on to children ?

**Comment by wakaztahir** [comment]
Sketcher only consumes the event if zoom is not enabled otherwise returns true in pointerInteropFilter !! 
tried returning false , doesn't work

**Comment by wakaztahir** [comment]
Could you please create a canvas and test panning and check why it might be failing ?

**Comment by wakaztahir** [comment]
So Canvas is not the problem , I have put the Zoomable inside the Bottom Sheet Content using Bottom Sheet Scaffold , took me a while to realize , I have another canvas inside zoomable whose panning works fine , The problem must be Bottom Sheet because that's the only difference between the two

**Comment by wakaztahir** [comment]
The problem is not Bottom sheet , I separated my composable from it and put it inside a dialog and still doesn't work !
The problem is unclear for now !

**Comment by wakaztahir** [comment]
## Problem 1
OK , I have zoom enabled as a mutable state in my view model , that goes into the zoomable
When I have set it to true in the view model , PANNING WORKS but if I set it to false , and set it to true again through a button PANNING does not work 

## Problem 2
When set to true initially (in view model)
- Zoom Works
- Panning Works
- Zoom Can't Be Disabled With A Button (Changing The State Only Disables Zoom and Panning Stays Enabled)
- It consumes the touch events (even when disabled afterwards using mutable state  , I want touch events when zoom disabled)

So basically enabled should work always , with the change of state and everything , panning should not consume touch events when zoom is disabled or maybe when panning is disabled

**Comment by wakaztahir** [comment]
Maybe there should be another state to control panning enabled !

**Comment by wakaztahir** [comment]
I can send zoomEnabled to true initially whish should solve my panning and zooming problem but setting it false again should disable panning and should not consume touch events so I can draw on canvas

**Comment by wakaztahir** [comment]
https://user-images.githubusercontent.com/42442700/124575302-b6ce4780-de64-11eb-9b8a-bfe1f31b66f2.mp4

enable (ZoomEnabled) is initially true (to make panning work)
Zoom button changes zoom enabled state
